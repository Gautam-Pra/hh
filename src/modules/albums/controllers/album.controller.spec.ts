import { beforeAll, describe, expect, it } from 'vitest'
import type { z } from 'zod'
import type { AlbumModel } from '#modules/albums/models'
import { AlbumController } from '#modules/index'

describe('AlbumController', () => {
  let albumController: AlbumController

  beforeAll(() => {
    albumController = new AlbumController()
    albumController.initRoutes()
  })

  it('retrieve album by link', async () => {
    const response = await albumController.controller.request(
      '/albums?link=https://www.jiosaavn.com/album/future-nostalgia/ITIyo-GDr7A_'
    )

    const wow = (await response.json()) as { data: z.infer<typeof AlbumModel> }

    expect(wow.data).toMatchSnapshot({
      playCount: expect.any(Number),
      songCount: expect.any(Number),
      songs: expect.arrayContaining([
        expect.objectContaining({
          playCount: expect.any(Number)
        })
      ])
    })
  })

  it('retrieve album by id', async () => {
    const response = await albumController.controller.request('/albums?id=23241654')

    const { data } = (await response.json()) as { data: z.infer<typeof AlbumModel> }

    expect(data).toMatchSnapshot({
      playCount: expect.any(Number),
      songCount: expect.any(Number),
      songs: expect.arrayContaining([
        expect.objectContaining({
          playCount: expect.any(Number)
        })
      ])
    })
  })
})
