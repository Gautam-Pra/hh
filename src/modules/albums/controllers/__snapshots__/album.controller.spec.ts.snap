// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlbumController > retrieve album by id 1`] = `
{
  "artists": {
    "all": [
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
    "featured": [],
    "primary": [
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
  },
  "description": "2020 路 English Album 路 Dua Lipa",
  "explicitContent": true,
  "id": "23241654",
  "image": [
    {
      "quality": "50x50",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-50x50.jpg",
    },
    {
      "quality": "150x150",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-150x150.jpg",
    },
    {
      "quality": "500x500",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-500x500.jpg",
    },
  ],
  "language": "english",
  "name": "Future Nostalgia",
  "playCount": Any<Number>,
  "songCount": Any<Number>,
  "songs": ArrayContaining [
    ObjectContaining {
      "playCount": Any<Number>,
    },
  ],
  "type": "album",
  "url": "https://www.jiosaavn.com/album/future-nostalgia/ITIyo-GDr7A_",
  "year": 2020,
}
`;

exports[`AlbumController > retrieve album by link 1`] = `
{
  "artists": {
    "all": [
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
    "featured": [],
    "primary": [
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
  },
  "description": "2020 路 English Album 路 Dua Lipa",
  "explicitContent": true,
  "id": "23241654",
  "image": [
    {
      "quality": "50x50",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-50x50.jpg",
    },
    {
      "quality": "150x150",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-150x150.jpg",
    },
    {
      "quality": "500x500",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-500x500.jpg",
    },
  ],
  "language": "english",
  "name": "Future Nostalgia",
  "playCount": Any<Number>,
  "songCount": Any<Number>,
  "songs": ArrayContaining [
    ObjectContaining {
      "playCount": Any<Number>,
    },
  ],
  "type": "album",
  "url": "https://www.jiosaavn.com/album/future-nostalgia/ITIyo-GDr7A_",
  "year": 2020,
}
`;
